### ISSUE TITLE
build failed: syntax error & unresolved references

### ISSUE THREAD

**Issue by briantoby** [kiyolee/openssl1_1-win-build#14]
Hi Kevin, I probably do not qualify in the category of "you know how to use VS, right?" but I am trying to build serf (ultimate goal is to build svn.exe) following the instructions you have at https://github.com/kiyolee/serf-win-build#readme but I am stuck building openSSL. I am using the community VS with a pretty minimal install of VS2022 components (in a Win10 VM, FWIW). First problem I had was that the build failed as it requires perl (worth noting in your README). Installed the Strawberry version and things went a lot farther, but now have two types of errors:

1st : many, many undefined externals, mostly beginning as BIO_, CRYPTO_, BN_, SSL_ or ASN1_, but probably others. 

2nd: syntax errors in openssl.c(!). For example, this line generates a bunch of errors:
```
static void list_type(FUNC_TYPE ft, int one);
```

I must be doing something wrong. Any suggestions would be much welcomed. 

Thanks, Brian

**Comment by kiyolee** [comment]
I think I do mention Perl in "Build Requirement".
Anyway, building subversion in my own way is also something I would also like to do.
However, there are other complexity (bigger ones) after serf, openssl, etc. At least for if I want to do it my way.
But for sure that'd a fun exercise if you would like to try.
So if you just want to *use* subversion, I'd recommend using prebuilt binaries instead.

As for the problems of building openssl1_1, can you share the build log?
I have never seen any problem as described.
But there are some common potential problems:
1. Using path with spaces and/or special characters. I guess that is a fair advice in general to NOT doing that for building software on Windows.
2. Any existing openssl installation (building in a different way or prebuilt version from somewhere else) getting into the way.
3. You need nasm too.

**Comment by briantoby** [comment]
This may be more than you want to know, but I am packaging some scientific software that needs python, various python packages and svn for Windows, Mac and Linux. Currently, I use anaconda, but that no longer includes all the packages I need anymore, so I need to migrate to conda-forge, which does not have svn for Windows. I worked on the windows svn recipe in conda-forge a bit. I suspect that someone who knows Windows builds better would be less frustrated with Azure, but I gave up. 

I am pretty sure I could modify my installer for Windows so that I include a svn.exe binary, rather than require it in a conda package. I have not looked closely at licenses, but I did not see any prebuilt svn dists that I would feel comfortable including in my own distributions, so my thought was do see if I could build my own version. That might also give me a head start on an Azure build as well. 

I had all put each of your git projects into c:\restart -- so no spaces. There is some openssl stuff in the strawberry perl dist, but I don't think any header files will be found by VS. I did have NASM in my VS config, though I probably would not have included it if I had a choice.  

Anyway, I am wondering if using "build solution" is my problem. In the interests of figuring out which build.log file had the errors, I opened the solution explorer and started building each project openssl1_1-win-build one-at-at time. They each worked. I went through the rest of the other git projects and they built OK.  I'm stumped as to what went wrong before.
